packages/flutter/lib/src/rendering/viewport.dart

```diff
@@ -311,7 +310,7 @@ abstract class RenderViewportBase<ParentDataClass extends ContainerParentDataMix
   /// When the style is [CacheExtentStyle.viewport], it is the main axis extent
   /// of the viewport multiplied by the requested cache extent, which is still
   /// expressed in pixels.
-  double _calculatedCacheExtent;
+  double? _calculatedCacheExtent;
 
```

Comment:
```suggestion

  // TODO(ianh): cacheExtent/cacheExtentStyle should be a single
  // object that specifies both the scalar value and the unit, not a
  // pair of independent setters. Changing that would allow a more
  // rational API and would let us make the getter non-nullable.

```